rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -357,6 +356,15 @@ def _extract_and_add_features(
                     f"don't coincide in '{message.text}' for attribute '{attribute}'."
                 )
 
+        if (
+            self.component_config[NUM_TRANSFORMER_LAYERS] == 0
+            and not self.component_config[ENTITY_RECOGNITION]
+            and attribute != INTENT
```

Comment:
I get where it is coming from, but could you please add comments why we add all of this conditions, so that we don't forget